### ISSUE TITLE
Maintain extra fields on upload

### ISSUE THREAD

**Issue by ivansg44** [Public-Health-Bioinformatics/covid19ValidationGrid#23]
Potential issue: If we insert the extra fields close to the same index they were originally, the first header row would become inaccurate. Could highlight fields to indicate the first header has no bearing on them. Not a super clean solution.

**Comment by ddooley** [comment]
I like the idea of flagging that a user column is a "custom" column by giving it a color in header.  Maybe we need a wee legend at bottom. 
field legend: [yellow] required    [violet]  recommended   [green]  custom user field

**Comment by ddooley** [comment]
later we could add functionality to enable users to drag them around but the fact that they placed them adjacent to others in a section is a positive in my UI book.

**Comment by ivansg44** [comment]
Good idea on the legend, but I think we should hide the legend in a "Help" menu on the top-left toolbar. The menu could also have a link to the SOP, and maybe the version information.

In the name of further decluttering the screen, it might be worth while moving the add rows functionality to the Settings menu too. However, Google Sheets has a similar add-rows button near the bottom of the spreadsheet.

**Comment by ivansg44** [comment]
I would also avoid using green, because yellow and green are a notably bad combination for colour-blind users.

**Comment by ddooley** [comment]
UI is a delicate balance indeed.  One thing I personally like to do is have the various elements that can be handy in a screen snapshot / diagram / printout visible on a line that one can choose to include or exclude during the screen grab.  E.g. if a bottom line has legend, date of screen capture, version #.  Others would say lets just create a switch to show/hide such elements in one go.  Re. #rows input, I can see why one might tuck that into a menu.  Its used maybe once during a work session.

**Comment by ddooley** [comment]
re. green, maybe you know of another pallet combo.  I'm guessing to address color blind users, switching to patterned backgrounds would be a future solution.

**Comment by ddooley** [comment]
I'll close this for now.  The new FieldMapper makes clear which fields are or are not mappable to a specification.  What I realize is that to accept new fields (to load and save) is to accept building a new schema off of an existing one.  DH should focus on a feature that incorporates data fields into a new specification (in SchemaEditor).
